alacritty_terminal/src/grid/mod.rs

```diff
@@ -427,6 +401,56 @@ impl<T> Grid<T> {
     }
 }
 
+/// Grid dimensions.
+pub trait Dimensions {
+    /// Total number of lines in the buffer, this includes scrollback and visible lines.
+    fn total_lines(&self) -> usize;
+
+    /// Height of the viewport in lines.
+    fn num_lines(&self) -> Line;
```

Comment:
I've kinda gotten used to `num_lines` but I've always thought it's a horrible name to settle on. Since there's really no way to know which one this represents. However it is extremely short which is important. `viewport_lines` is significantly longer.

Maybe `visible_lines` would be better, it's at least a little shorter. The name `screen_lines` would be even shorter.

I've decided on `screen_lines` or now, let me know what you think. I've also renamed `num_cols` to just `cols`, since we don't have `num_` as a prefix for the other functions either.